www/js/lib/zimfile.js

```diff
@@ -48,89 +47,94 @@ define(['xzdec_wrapper', 'zstddec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'],
      * @property {Integer} mimeListPos position of the MIME type list (also header size)
      * @property {Integer} mainPage main page or 0xffffffff if no main page
      * @property {Integer} layoutPage layout page or 0xffffffffff if no layout page
-     * 
      */
     
     /**
-     * @param {Array.<File>} abstractFileArray
+     * Abstract an array of one or more (split) ZIM archives
+     * @param {Array<File>} abstractFileArray An array of ZIM file parts
      */
-    function ZIMFile(abstractFileArray)
-    {
+    function ZIMFile(abstractFileArray) {
         this._files = abstractFileArray;
     }
 
     /**
-     * 
-     * @param {Integer} offset
-     * @param {Integer} size
-     * @returns {Integer}
+     * Read and decode an integer value from the ZIM archive
+     * @param {Integer} offset The offset at which the integer is found
+     * @param {Integer} size The size of data to read
+     * @returns {Promise<Integer>} A Promise for the returned value 
      */
-    ZIMFile.prototype._readInteger = function(offset, size)
-    {
-        return this._readSlice(offset, size).then(function(data)
-        {
+    ZIMFile.prototype._readInteger = function (offset, size) {
+        return this._readSlice(offset, size).then(function (data) {
             return readInt(data, 0, size);
         });
     };
 
     /**
-     * 
-     * @param {Integer} offset
-     * @param {Integer} size
-     * @returns {Promise}
+     * Read a slice from the ZIM set starting at offset for size of bytes
```

Comment:
It might be added in this description that it also makes use of the cache

Reply:
Will do.